### ISSUE TITLE
ENH: stats: T Testing of lognormal distribution

### ISSUE THREAD

**Issue by edo-cnl** [scipy/scipy#22171]
### Is your feature request related to a problem? Please describe.

Scipy has a function for 1 sample T-Testing (https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_1samp.html) , however if the data is log normally distributed the function does not help, as it is erroneous to test a distribution transformed with log against the log of a point value popmean.

### Describe the solution you'd like.

A new function that performs a T test for a lognormal distributions against the alternative hypothesis, when the non-transformed popmean takes a given value.

Given the standard parameters for a T test function ( sample, popmean, alpha, [two-sided, upper, lower] ), the function should:
1) apply log (or any other transformation) to the sample 
2) calculate the CI(s) at alpha significance for the log distribution
3) invert the CI(s) with exponential
4) compare the CI(s) with popmean, and yield the result

**Comment by mdhaber** [comment]
> it is erroneous to test a distribution transformed with log against the log of a point value popmean.

If I understand correctly, this is simply because `log(mean(X)) != mean(log(X))`, where the left hand side is "the log of a point value popmean".

But would it be wrong to pass `ttest_1samp` the log of the observations and "the population mean of the log of the distribution" (`mean(log(X))`)?

Is the problem that you don't know `mean(log(X))`, only `mean(X)`, where `X` is lognormally distributed?

**Comment by edo-cnl** [comment]
> Is the problem that you don't know mean(log(X)), only mean(X), where X is lognormally distributed? (In "when the non-transformed popmean takes a given value", I would have thought the "non-transformed popmean" would be that of the underlying normal distribution, but perhaps that is not intended.)

Correct.
Popmean is the name of the parameter for `ttest_1samp`, but can be any value.

In my case, I have a sample that proved to be lognormal, and a value (the popmean) to test it against.
You are also correct that in point 3 of the description, the scale of the original distribution should be used for the inversion.
However, since the distribution is unknown, I resort to using the st_dev of the log transformed sample.

I had resorted to this  
```
def one_tailed_t_test_lognormal_distribution_against_point_value(
    log_data: pd.Series,
    any_value: float,
    alternative="sample_mean_significantly_lower",
    alpha=0.05,
) -> bool:
    """
    one_tailed_t_test_lognormal_distribution_against_point_value
    Args:
        log_data (pd.Series): Series of log transformed data that is normally distributed
        any_value (float): a value to test the log transformed series against
        alternative (str): side to test: lognormal distribution is either 'sample_mean_significantly_lower'
            or 'sample_mean_significantly_higher' than the passed atv
        alpha (float): significance level
    Returns:
        bool: whether each mid's transactions mean significantly deviates from the passed atv
            , in the specified direction
      """

    if alternative not in [
        "sample_mean_significantly_lower",
        "sample_mean_significantly_higher",
    ]:
        raise ValueError(
            """alternative must be either 'sample_mean_significantly_lower'
                            or 'sample_mean_significantly_higher'"""
        )

    sample_mean = log_data.mean()
    sample_std_dev = log_data.std()
    sample_size = len(log_data)
    std_error = sample_std_dev / np.sqrt(sample_size)

    if alternative == "sample_mean_significantly_lower":
        log_upper_bound = scipy.stats.norm.ppf(1 - alpha, sample_mean, std_error)
        upper_bound = np.exp(log_upper_bound + sample_std_dev**2 / 2)
        return upper_bound < any_value

    elif alternative == "sample_mean_significantly_higher":
        log_lower_bound = scipy.stats.norm.ppf(alpha, sample_mean, std_error)
        lower_bound = np.exp(log_lower_bound + sample_std_dev**2 / 2)
        return lower_bound > any_value
```

**Comment by mdhaber** [comment]
>  I resort to using the st_dev of the log transformed sample.

Well, sure, but if you do that, is it any different than passing `ttest_1samp` the corresponding `popmean`? More explicitly:
```
x = ... # lognormally distributed data
lognorm_popmean = ... # hypothesized population mean
y = np.log(x)
s = np.std(y, ddof=1)  # probably sould include the bias adjustment, no?
popmean = lognorm_popmean - s**2/2
```

In any case, I assumed that might be what you had in mind. I tested some similar ideas, and the coverage of the confidence interval under the null hypothesis was not close to the desired level. Here is how you'd perform such a test with your idea:

```python3
import numpy as np
from scipy import stats
rng = np.random.default_rng(945843958324953)

n, m = 10000, 20  # 10000 samples, each with 20 observations
alpha = 0.95  # desired coverage
x = rng.standard_normal(size=(n, 20))
low, high = stats.ttest_1samp(x, popmean=0, axis=-1).confidence_interval(alpha)
count = np.count_nonzero((low < 0) &  (0 < high))
print(count/n)  # 0.951
```

If the coverage is close, we can investigate further to find a mathematical argument for this procedure.

**Comment by edo-cnl** [comment]
I tested the procedure in a notebook ( https://github.com/edo-cnl/edo-playground/blob/main/experiment_lognormality_inversion_with_sample_mean.ipynb ) and I find the coverage to be promising:
- average accuracy > 95% even for small sample sizes,
- accuracy stably consistently above 95% as sample size exceeds 2000.
- bias reduction through bootstrapping seems effective in improving accuracy, however its beneficial effects decrease as sample size enlarges.


>  is it any different than passing ttest_1samp ?

I see no difference, but I see value in having a one-liner function for it.

**Comment by mdhaber** [comment]
> average accuracy > 95% even for small sample sizes,

Existing hypothesis tests in SciPy almost always produce a p-value, and sometimes also produce a confidence interval. You can use those results to perform a classification, but SciPy doesn't produce a binary result. Which did you intend for this function to return?

If this will return a p-value, we want to make sure the p-value is uniformly distributed under the null hypothesis, and if a confidence interval, we'll need to see that the coverage is right regardless of the `confidence_level`. Can you rephrase your results in terms of those metrics rather than accuracy?

**Comment by edo-cnl** [comment]
> we tend not to invent approximate tests and add them as convenience methods
Very fair.
Given the popularity of the lognormal, it seems absurd to me that a robust way to perform parametric hypothesis testing has not been found.
Let me do some more research on this.

Will close this issue and potentially open a new one.
Thanks for your reviews !
